events/include/gz/common/Event.hh

```diff
@@ -93,39 +151,37 @@ namespace gz
 
       /// \brief Friend class.
       public: template<typename T, typename N> friend class EventT;
+      public: friend class Event;
     };
 
     /// \brief A class for event processing.
     /// \tparam T function event callback function signature
     /// \tparam N optional additional type to disambiguate events with same
     ///   function signature
+    //
+    // WARNING: This class should not have any data members. Instead all data
```

Comment:
Are we sure that no one is currently doing this?

Reply:
Yes, I was able to compile all gz libraries without any issues. But as I mentioned in the test, users that call `delete` through a base pointer might get a compiler warning saying that `Event` doesn't have a virtual destructor. This warning doesn't occur when using a smart pointer though.